Implement X11 pasting

Davipb opened the pull request:
This PR implements pasting for X11, meaning that it'll be possible to paste text or files copied from `clipboard` into other applications.

This is done by forking `clipboard` and using the fork as a paste daemon that replies to X11 selection requests until another application steals the clipboard. The daemon exits automatically when it has no more work to do.

Slackadays commented:
I like this, but I noticed that there's a `const` back in the specializations for `ClipboardContent` which I had to remove to get Clang to compile with it on macOS. The reaskn for that was because it couldn't guarantee that passing the value of the function to the macOS C wrappers wouldn't change, because the C functions expect something that is always constant 100% of the time. Did you account for this?

Davipb commented:
Do you mean in the constructors like `ClipboardContent(std::string const& text)`? I added it mostly as a way to prevent us from using `std::move()` on the values mistakenly (which was being done before, and I also removed in the PR), since that can cause random objects to just lose their data if we're not careful. 

If this is just a technicality when passing things to C, we can always `const_cast` it away while keeping the C++ code safe.

Slackadays commented:
Let's see what happens if we merge it since I don't have access to my MacStadium Mac mini right now to test with.
